mono/metadata/mono-security-windows-uwp.c

```diff
@@ -41,8 +41,8 @@ ves_icall_System_Security_Principal_WindowsIdentity_GetRoles (gpointer token)
 {
 	// FIXME This is now supported by UWP.
 	ERROR_DECL (error); // FIXMEcoop
-	mono_security_win_not_supported ("GetTokenInformation", error);
-	mono_error_set_pending_exception (error); // FIXMEcoop
+	mono_security_win_not_supported ("GetTokenInformation", &error);
```

Comment:
This doesn't look correct. It should not compile. `ERROR_DECL` makes `MonoError* error`.

Reply:
I does compile, strange, will look into why that happened.

Reply:
The C compilers are very lax that way, not surprising. There might be a warning.

Reply:
So the above fails to compile under C++ compiler (as expected), but under C compiler it works, but will give incorrect runtime result. There is a warning around constness, warning C4090: 'function': different 'const' qualifiers, but that warning is currently disabled in winconfig.h so it didn't show up. The other warning that normally should popup is warning C4047: 'operator' : 'identifier1' differs in levels of indirection from 'identifier2' but that is currently also disabled in winconfig.h so that's why this didn't show up during compile. I believe we should see if we could re-enable and fix some of these warnings (or disable on a granular level) into the since they are level 1 warnings. Thanks for spotting this.